### PR TITLE
Drop use of node core streams from socket -> handler -> socket path

### PR DESCRIPTION
**Pull request by jcorbin**
This is between a 20% and 30% speedup on our synthetic benchmark.

At the cost of dropping `NODE_DEBUG=tchannel_dump` convenience for now.

cc @Raynos @kriskowal

### CODE REVIEW

File: node/index.js [hunk 1]
```diff
@@ -682,34 +684,43 @@ TChannelConnection.prototype.setupHandler = function setupHandler() {
...
+    function handleReadFrame(frame) {
         if (!self.closing) {
             self.lastTimeoutTime = 0;
         }
+        self.handler.handleFrame(frame, hanldedFrame);
     }
 
-    function onReaderError(err) {
-        self.onReaderError(err);
+    function hanldedFrame(err) {
```
Comment (kriskowal) [inline node/index.js]: mis-spell

**Comment by kriskowal** [comment]
Great!

File: node/index.js [hunk 2]
```diff
@@ -785,21 +790,6 @@ TChannelConnection.prototype.close = function close(callback) {
     self.socket.destroy();
 };
 
-TChannelConnection.prototype.onReaderError = function onReaderError(err) {
```
Comment (Raynos) [inline node/index.js]: this method is still used.

File: node/index.js [hunk 3]
```diff
@@ -739,12 +750,6 @@ TChannelConnection.prototype.setupHandler = function setupHandler() {
         // TODO: should terminate corresponding inc res
     }
 
-    function onHandlerError(err) {
```
Comment (Raynos) [inline node/index.js]: function still used.

Comment (Raynos) [inline node/index.js]: I see; this function moved. fair enough.

**Comment by Raynos** [comment]
seems legit.

**Pull request merged**